### PR TITLE
DALI proxy examples

### PR DESCRIPTION
**Pull request by jantonguirao**
## Category:

**Other**

## Description:

Adds L3 tests and code samples for RN50 and EfficientNet with DALI proxy
Depends on https://github.com/NVIDIA/DALI/pull/5726

### Tests:

- [ ] Existing tests apply
- [x] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [x] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [x] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [x] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [ ] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: docs/examples/use_cases/pytorch/efficientnet/image_classification/dali.py [hunk 1]
```diff
@@ -64,22 +70,32 @@ def training_pipe(data_dir, interpolation, image_size, output_layout, automatic_
                                       mean=[0.485 * 255, 0.456 * 255, 0.406 * 255],
                                       std=[0.229 * 255, 0.224 * 255, 0.225 * 255])
 
-    return output, labels
+    if labels:
+        return output, labels
+    else:
+        return output
+
```
Comment (szkarpinski) [inline docs/examples/use_cases/pytorch/efficientnet/image_classification/dali.py:77]: What condition can make `labels` none? Can we express that explicitly here?
```
if actual_condition:
   return output, labels
else:
   return output
```

File: docs/examples/use_cases/pytorch/efficientnet/image_classification/dali.py [hunk 2]
```diff
@@ -64,22 +70,32 @@ def training_pipe(data_dir, interpolation, image_size, output_layout, automatic_
...
+    decoder_device = "mixed" if dali_device == "gpu" else "cpu"
+    images = fn.decoders.image(jpegs, device=decoder_device, output_type=types.RGB)
 
-    images = fn.decoders.image(jpegs, device="mixed", output_type=types.RGB)
+    images = fn.resize(images, resize_shorter=image_size,
+                       interp_type=interpolation, antialias=False)
 
-    images = fn.resize(images, resize_shorter=image_size, interp_type=interpolation,
-                       antialias=False)
+    images = images.gpu()
```
Comment (szkarpinski) [inline docs/examples/use_cases/pytorch/efficientnet/image_classification/dali.py:91]: Did I read it right that before the validation pipeline was returning CPU data, and now it's GPU? Why?

File: docs/examples/use_cases/pytorch/efficientnet/image_classification/training.py [hunk 3]
```diff
@@ -211,13 +211,44 @@ def train(
...
+
+            end = time.time()
+            if prof > 0 and (i + 1 >= prof):
+                time.sleep(5)
+                break
+            if ((i + 1) % 20 == 0) and timeout_handler.interrupted:
+                time.sleep(5)
+                interrupted = True
+                break
+        return interrupted
```
Comment (szkarpinski) [inline docs/examples/use_cases/pytorch/efficientnet/image_classification/training.py:250]: This is a copy-paste from some other part of the example, right? Can we deduplicate it?

File: docs/examples/use_cases/pytorch/efficientnet/image_classification/utils.py [hunk 4]
```diff
@@ -181,3 +181,34 @@ def calc_ips(batch_size, time):
...
+    elif isinstance(obj, (list, tuple, set)):
+        for item in obj:
+            found.extend(find_children_by_type_name(item, target_type_name))
+
+    # If the object has a __dict__, check its attributes (for custom objects)
+    elif hasattr(obj, "__dict__"):
+        for attr_value in vars(obj).values():
+            found.extend(find_children_by_type_name(attr_value, target_type_name))
+
+    return found
```
Comment (szkarpinski) [inline docs/examples/use_cases/pytorch/efficientnet/image_classification/utils.py:214]: Is this really needed to use DALI Proxy? Can we avoid that? It's quite scary ;)

File: docs/examples/use_cases/pytorch/resnet50/main.py [hunk 5]
```diff
@@ -345,44 +424,57 @@ def resume():
...
+                        prec1,
+                        prec5,
+                        args.total_batch_size / total_time.avg))
 
 class data_prefetcher():
     """Based on prefetcher from the APEX example
        https://github.com/NVIDIA/apex/blob/5b5d41034b506591a316c308c3d2cd14d5187e23/examples/imagenet/main_amp.py#L265
     """
-    def __init__(self, loader):
+    def __init__(self, loader, normalize):
```
Comment (szkarpinski) [inline docs/examples/use_cases/pytorch/resnet50/main.py:471]: Is this option needed to demonstrate DALI Proxy? I guess it was useful for benchmarking, but would it be useful to the user/reader of this example?

File: docs/examples/use_cases/pytorch/resnet50/main.py [hunk 6]
```diff
@@ -425,15 +518,17 @@ def train(train_loader, model, criterion, scaler, optimizer, epoch):
     # switch to train mode
     model.train()
     end = time.time()
+    normalize = not args.dali_proxy
```
Comment (mzient) [inline docs/examples/use_cases/pytorch/resnet50/main.py:521]: Why?

File: docs/examples/use_cases/pytorch/efficientnet/image_classification/utils.py (hunk 4 above)
Comment (jantonguirao) [reply docs/examples/use_cases/pytorch/efficientnet/image_classification/utils.py:214]: not really need. Let me see if I can come up with something better
